src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -35,11 +35,12 @@ public FindAllReferencesHandler(IMetadataAsSourceFileService metadataAsSourceFil
             Solution solution,
             ReferenceParams referenceParams,
             ClientCapabilities clientCapabilities,
+            string? clientName,
             CancellationToken cancellationToken)
         {
             Debug.Assert(clientCapabilities.HasVisualStudioLspCapability());
 
-            var document = solution.GetDocumentFromURI(referenceParams.TextDocument.Uri);
+            var document = solution.GetDocumentFromURI(referenceParams.TextDocument.Uri, clientName);
```

Comment:
Do we have to do any filtering for this one? Since if you have a search from outside a razor document that ends up going _into_ a razor document does that need to exclude those hits?

Reply:
We do I believe - this document is just the one containing the trigger location - so we don't want both language servers to trigger on non-razor documents (we'd get entirely duplicated results).  But that doesn't prevent a find refs triggered from the non razor LSP from providing results inside razor documents.